### DISCUSSION TITLE
Can an alert rule template be configured for a single machine?

### DISCUSSION THREAD

**Discussion by czxin788** [ccfos/nightingale#2002] [Ideas]
夜莺的告警规则配置方法和zabbix是有很大区别的。

zabbix是将模版配置给单个机器，比如有个机器用zabbix监控，可以单独给这个机器绑定linux监控模版，mysql监控模版等。这样的好处是，如果我想单独调整一下这个机器的告警阈值，比如模版是大于80%内存使用就告警，那我可以单独对这个机器禁用这个80%内存告警，然后再对这个机器单独新建一个大于90%的内存告警。

再看夜莺的告警规则，它是针对业务组生效的，业务组下有好多机器。这么做的缺点就是单独调整某个机器的告警策略，就没有zabbix那样灵活。

对于夜莺的这个问题困扰我很多年了，不知道是不是我的使用方法有问题。

**Comment by UlricQin** [comment]
夜莺目前的告警逻辑是承袭自 Prometheus 生态。在 Prometheus 生态里，不同的机器有不同的告警阈值是通过 promql 来过滤，比如你所有的机器的 cpu_usage_idle 小于 20 告警，但是唯独有一台机器（假设是 hosta）要求小于 30 告警，应该设置两个告警规则，比如：

```
cpu_usage_idle{ident!="hosta"} < 20
cpu_usage_idle{ident="hosta"} < 30
```

当然，未必一定使用 ident 标签过滤，具体可以使用什么标签过滤，取决于 cpu_usage_idle 这个指标有哪些标签。

之所以出现和 zabbix 这样的不同设计，是因为 zabbix 更侧重机器设备监控，偏资产管理的视角，Prometheus 生态的设计更偏应用监控，倾向于一次配置一批 series，通过标签做过滤。后面夜莺也有计划提供类似 zabbix 这样的能力，不过人力问题，目前还没有排期。

**Comment by czxin788** [comment]
谢谢回复，明白原因了
